ui/imports/shared/views/AssetsDetailView.qml

```diff
@@ -18,8 +19,78 @@ Item {
+        property real maxStep: 22000
+
+        property var graphTabsModel: [{text: "Price", enabled: true}, {text: "Balance", enabled: false}]
+        property var timeRangeTabsModel: [{text: "1H", enabled: true},
+                {text: "1D", enabled: true},{text: "7D", enabled: true},
+                {text: "1M", enabled: true}, {text: "6M", enabled: true},
+                {text: "1Y", enabled: true}, {text: "ALL", enabled: true}]
+
+        property var simTimer: Timer {
+            running: true
+            interval: 3000
+            repeat: true
+            onTriggered: {
+                d.generateData();
+            }
+        }
+
+        function minutes(minutes = 0) {
+            var newMinute = new Date(new Date().getTime() - (minutes * 60 * 1000)).toString();
+            if (newMinute.slice(10,12) === "00") {
+                var dateToday = new Date(Date.now()).toString();
+                return dateToday.slice(4,7) + " " + dateToday.slice(8,10);
+            }
+            return newMinute.slice(10,16);
+        }
+
+        function hour(hours = 0) {
+            var newHour = new Date(new Date().getTime() - (hours * 60 * 60 * 1000)).toString();
+            if (newHour.slice(10,12) === "00") {
+                var dateToday = new Date(Date.now()).toString();
+                return dateToday.slice(4,7) + " " + dateToday.slice(8,10);
+            }
+            return newHour.slice(10,16);
+        }
+
+        function day(before = 0) {
+            var newDay = new Date(Date.now() - before * 24 * 60 * 60 * 1000).toString();
+            return newDay.slice(4,7) + " " + newDay.slice(8,10);
+        }
+
+        function month(before = 0) {
+            var newMonth = new Date(Date.now() - before * 24 * 60 * 60 * 1000).toString();
+            return newMonth.slice(4,7) + " '" + newMonth.slice(newMonth.indexOf("G")-3, newMonth.indexOf("G")-1);
+        }
+
+        property var timeRange: [
+            {'1H': [minutes(60), minutes(55), minutes(50), minutes(45), minutes(40), minutes(35), minutes(30), minutes(25), minutes(20), minutes(15), minutes(10), minutes(5), minutes()]},
+            {'1D': [hour(24), hour(23), hour(22), hour(21), hour(20), hour(19), hour(18), hour(17), hour(16), hour(15), hour(14), hour(13),
+                    hour(12), hour(11), hour(10), hour(9), hour(8), hour(7), hour(6), hour(5), hour(4), hour(3), hour(2), hour(1), hour()]},
+            {'7D': [day(6), day(5), day(4), day(3), day(2), day(1), day()]},
+            {'1M': [day(30), day(28), day(26), day(24), day(22), day(20), day(18), day(16), day(14), day(12), day(10), day(8), day(6), day(4), day()]},
+            {'6M': [month(150), month(120), month(90), month(60), month(30), month()]},
+            {'1Y': [month(330), month(300), month(270), month(240), month(210), month(180), month(150), month(120), month(90), month(60), month(30), month()]},
+            {'ALL': ['2016', '2017', '2018', '2019', '2020', '2021', '2022']}
```

Comment:
Should "2022" be hardcoded like that? Shouldn't it go all the way to the "current" year?

Reply:
it's dummy data, I assume all these will be removed anyway when the backend part is ready, this is why i didn't put a lot of effort on this

Reply:
Fair enough